restapi/src/main/java/io/stargate/web/docsapi/service/query/ExpressionParser.java

```diff
@@ -19,24 +19,34 @@
 import com.bpodgursky.jbool_expressions.And;
 import com.bpodgursky.jbool_expressions.Expression;
 import com.bpodgursky.jbool_expressions.Literal;
+import com.bpodgursky.jbool_expressions.Or;
 import com.bpodgursky.jbool_expressions.rules.RuleSet;
 import com.fasterxml.jackson.databind.JsonNode;
+import io.stargate.web.docsapi.exception.ErrorCode;
+import io.stargate.web.docsapi.exception.ErrorCodeRuntimeException;
 import io.stargate.web.docsapi.service.query.condition.BaseCondition;
 import io.stargate.web.docsapi.service.query.condition.ConditionParser;
 import io.stargate.web.docsapi.service.util.DocsApiUtils;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
 import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
+import java.util.Objects;
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 import javax.inject.Inject;
 import javax.ws.rs.core.PathSegment;
+import org.apache.commons.lang3.mutable.MutableInt;
 
 public class ExpressionParser {
 
+  private static final String OR_OPERATOR = "$or";
+
+  private static final String AND_OPERATOR = "$and";
```

Comment:
I think if we introduce `OR` we ought to support `NOT` too, for the same of completeness... WDYT?

Reply:
Pufff, I would keep this separated.. Also I am not sure how would we process this atm.. Additional issue?

Reply:
ok, pls file another issue.

Reply:
https://github.com/stargate/stargate/issues/1071